cli/args/flags.rs

```diff
@@ -2665,7 +2670,15 @@ fn test_parse(flags: &mut Flags, matches: &clap::ArgMatches) {
 
   let concurrent_jobs = if matches.is_present("jobs") {
     if let Some(value) = matches.value_of("jobs") {
+      println!(
+          "{}",
+          crate::colors::yellow("Warning: --jobs flag with numeric value is deprecated. Use 'DENO_JOBS' environment variable with --jobs flag instead."),
+        );
```

Comment:
One scenario I'm wondering about is when people really want to limit the parallelism to say `2` due to the tests using some common resource that needs to be limited. I think probably doing so is probably an anti-pattern because it affects all the tests and not just the ones that use the common resource. In this case, the resource usage should be limited by the tests themselves or the user should run the command in a deno task setting the DENO_JOBS env var first (ex. `DENO_JOBS=2 deno test`).